### PR TITLE
Fix for action releases and changelog updates

### PR DESCRIPTION
**Pull request by sumukhballal**
**Description of changes:**

1. Fixed and tested `install-dist` step in the release.yml file.
2. Updated CHANGELOG.md to include latest changes for release.

**Testing done:**

Tested  `install-dist` works as expected on my local branch build. https://github.com/sumukhballal/twoliter/actions/runs/10889775968/job/30217129584

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by bcressey — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,16 +9,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+## [0.4.6] - 2024-09-16
 
 ### Changes
 
 - Add support for vendor override files ([#344])
 - Updated buildsys to add new 'build-all' target, reduce build time ([#345], [#357])
 - CICD, workspace and doc improvements ([#353], [#354], [#355], [#358])
-- Add support for partial lockfile validation & refactor lock interfaces ([#361], [#363])
+- Add support for partial lockfile validation & refactor lock interfaces, improve logging ([#361], [#363], [#370])
 - Update tough dependencies to latest versions ([#365])
```
Comment (cbgbt) [inline CHANGELOG.md:20]:
```suggestion
- Add support for partial lockfile validation & refactor lock interfaces, improve logging ([#361], [#363], [#370])
- Add support for publishing kits to repositories that do not share a name with the kit ([#372])
- Update tough dependencies to latest versions ([#365])
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [#361]: https://github.com/bottlerocket-os/twoliter/pull/361
 [#363]: https://github.com/bottlerocket-os/twoliter/pull/363
 [#365]: https://github.com/bottlerocket-os/twoliter/pull/365
+[#369]: https://github.com/bottlerocket-os/twoliter/pull/369
+[#370]: https://github.com/bottlerocket-os/twoliter/pull/370
```
Comment (cbgbt) [inline CHANGELOG.md]:
```suggestion
[#370]: https://github.com/bottlerocket-os/twoliter/pull/370
[#372]: https://github.com/bottlerocket-os/twoliter/pull/372
```

**Review by cbgbt — APPROVED** [review]
Mind adding #372 to the changelog as well?

**Comment by sumukhballal** [comment]
Addressed @cbgbt 's comments.

**Review by cbgbt — APPROVED** [review]

**Pull request merged**